crates/proof-of-sql/src/base/database/slice_operation.rs

```diff
@@ -3,6 +3,19 @@ use alloc::{format, vec::Vec};
 use core::fmt::Debug;
 use num_traits::ops::checked::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub};
 
+/// Reverse a binary operator. That is, $a *_{op} b = b * a$.
+///
+/// With this function we don't need to consider the case of applying
+/// a binary operator to a single value and a slice, because we can
+/// always use `reverse_op` to reverse the order of the arguments.
+pub(crate) fn reverse_op<F, T>(op: F) -> impl Fn(&T, &T) -> T
+where
+    F: Fn(&T, &T) -> T,
+    T: Copy,
```

Comment:
Is the `T: Copy` needed?
```suggestion
pub(crate) fn reverse_op<F, S, T, U>(op: F) -> impl Fn(&S, &T) -> U
where
    F: Fn(&S, &T) -> U,
```